superdesk/io/feed_parsers/belga/belga_newsml_1_2.py

```diff
@@ -0,0 +1,452 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.appendsourcefabric.org/superdesk/license
+
+import datetime
+
+from superdesk.io.feed_parsers.newsml_1_2 import NewsMLOneFeedParser
+from superdesk.io.registry import register_feed_parser
+from superdesk.errors import ParserError
+from superdesk.etree import etree
+
+
+class BelgaNewsMLOneFeedParser(NewsMLOneFeedParser):
+    """
+    Feed Parser which can parse if the feed is in NewsML format, specific AFP, ANP, .. Belga xml.
+    """
+    NAME = 'belganewsml12'
+
+    label = 'Belga News ML 1.2 Parser'
+
+    def can_parse(self, xml):
+        return xml.tag == 'NewsML'
+
+    def parse(self, xml, provider=None):
+        """
+        Parser content the xml newsml file to json object.
+
+        Example content the xml newsml file:
+
+        <?xml version="1.0" encoding="utf-8"?>
+        <NewsML Version="1.2">
+          <!--AFP NewsML text-photo profile evolution2-->
+          <!--Processed by Xafp1-4ToNewsML1-2 rev21-->
+          <Catalog Href="http://www.afp.com/dtd/AFPCatalog.xml"/>
+          <NewsEnvelope>
+            ......
+          </NewsEnvelope>
+          <NewsItem xml:lang="fr">
+            <Identification>
+                .......
+            </Identification>
+            <NewsManagement>
+                ......
+            </NewsManagement>
+            <NewsComponent>
+                ......
+            </NewsComponent>
+          </NewsItem>
+        </NewsML>
+
+        :param xml:
+        :param provider:
+        :return:
+        """
+
+        try:
+            l_item = []
+            self.root = xml
+
+            # parser the NewsEnvelope element
+            item_envelop = self.parser_newsenvelop(xml.find('NewsEnvelope'))
+
+            # parser the NewsItem element
+            l_newsitem_el = xml.findall('NewsItem')
+            for newsitem_el in l_newsitem_el:
+                item = item_envelop
```

Comment:
I think it should use `item_envelop.copy()` or something, otherwise it would be updating same dict all the time

Reply:
Yeah, that right. Thanks.